src/PostgREST/Logger.hs

```diff
@@ -90,6 +90,9 @@ observationLogger loggerState logLevel obs = case obs of
   o@(HasqlPoolObs _) -> do
     when (logLevel >= LogDebug) $ do
       logWithZTime loggerState $ observationMessage o
+  o@(DBQuery sql status) -> do
+    when (sql /= mempty && shouldLogResponse logLevel status) $ do
```

Comment:
When is the `sql` mempty?

Reply:
Ah, that's for the OpenAPI output. This shouldn't be needed once `postgrest-openapi` replaces it. Yeah it needs a clarification there.

Reply:
Oh, this also affects the `OPTIONS` requests too. They do not call the database whatsoever, so `mempty` is expected for the SQL query.